src/EventTickets/Actions/RegisterEventsMenuItem.php

```diff
@@ -26,6 +29,18 @@ public function __invoke()
      */
     public function render()
     {
+        if(isset($_GET['id'])) {
+            $event = Event::find(absint($_GET['id']));
+
+            if (!$event) {
+                wp_die(__('Event not found', 'give-event-tickets'), 404);
```

Comment:
As a note, we can probably improve this user experience by redirecting to the list table when an event does not exist. The tricky part here is that we are mid request and may not be able to do a server side redirect, but we could localize some self instantiating javascript to change the window location.

Reply:
What we have here is fine for now, but something to consider in a separate PR.